common/statistics/src/clients/nym_api_statistics.rs

```diff
@@ -0,0 +1,77 @@
+//! # API Connection statistics
+//!
+//! Metrics collected by the client while attempting to pull config from the API.
+
+use super::ClientStatsEvents;
+use std::collections::VecDeque;
+
+use nym_metrics::{inc, inc_by};
+use serde::{Deserialize, Serialize};
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub(crate) struct NymApiStats {
+    // Sent
+    real_packets_sent: u64,
+    real_packets_sent_size: usize,
+
+    /// API connection failure statistics
+    failures: VecDeque<()>, // TODO
+}
+
+impl NymApiStats {
+    fn handle(&mut self, event: NymApiStatsEvent) {
+        match event {
+            NymApiStatsEvent::RealPacketSent(packet_size) => {
+                self.real_packets_sent += 1;
+                self.real_packets_sent_size += packet_size;
+                inc!("real_packets_sent");
+                inc_by!("real_packets_sent_size", packet_size);
+            }
+        }
+    }
+
+    fn summary(&self) -> (String, String) {
+        (
+            format!("packets sent: {}", self.real_packets_sent,),
+            String::new(),
```

Comment:
again, some strongly typed return type would definitely be useful here...

Reply:
Pinging @jmwample for this one